src/SongManager.cpp

```diff
@@ -422,12 +422,11 @@ void SongManager::LoadSongDir( RString sDir, LoadingWindow *ld, bool onlyAdditio
 		ld->SetTotalWork( songCount );
 	}
 
-	groupIndex = 0;
 	songIndex = 0;
-	
-	for (RString const &sGroupDirName : arrayGroupDirs)	// foreach dir in /Songs/
+	for (auto const &pair : mapGroupSongDirs)	// foreach dir in /Songs/

```

Comment:
Use structured bindings:

`for (const auto& [sGroupDirName, arraySongDirs] : mapGroupSongDirs)`